`qml.compile` docstring clarification

isaacdevlugt opened the pull request:
**Context:** `pipeline` behaviour / implications wasn't exactly clear w.r.t. the default set of transforms that get applied

**Description of the Change:** Docstring update

**Benefits:** much benefits

**Possible Drawbacks:** 🙅 

**Related GitHub Issues:**

isaacdevlugt committed:
basis rotation code exampleg

isaacdevlugt committed:
Merge branch 'master' of https://github.com/PennyLaneAI/pennylane

isaacdevlugt committed:
Merge branch 'master' of https://github.com/PennyLaneAI/pennylane

isaacdevlugt committed:
Merge branch 'master' of https://github.com/PennyLaneAI/pennylane

isaacdevlugt committed:
Merge branch 'master' of https://github.com/PennyLaneAI/pennylane

isaacdevlugt committed:
Merge branch 'master' of https://github.com/PennyLaneAI/pennylane

isaacdevlugt committed:
compile dostring

isaacdevlugt commented on doc/releases/changelog-dev.md:
```diff
@@ -256,6 +259,10 @@
 * Fixed a typo in the code example for `qml.labs.dla.lie_closure_dense`.
   [(#6858)](https://github.com/PennyLaneAI/pennylane/pull/6858)
 
+* The code example in the docstring for `qml.BasisRotation` was corrected by including `wire_order` in the 
```
huh

isaacdevlugt commented on pennylane/templates/subroutines/basis_rotation.py:
```diff
@@ -68,7 +68,7 @@ class BasisRotation(Operation):
         ...    qml.BasisRotation(wires=wires, unitary_matrix=umat)
-        >>> circ_unitary = qml.matrix(circuit)()
+        >>> circ_unitary = qml.matrix(circuit, wire_order=wires)()
         >>> np.round(circ_unitary/circ_unitary[0][0], 3)
```
huh

isaacdevlugt committed:
pseudo merge master

isaacdevlugt committed:
changelog

isaacdevlugt commented on doc/releases/changelog-dev.md:
```diff
@@ -244,6 +244,9 @@
 
 <h3>Documentation 📝</h3>
 
+* The definitition of the `pipeline` argument for `qml.compile` was clarified in
+  its docstring.
```
```suggestion
* The definitition of the `pipeline` argument for `qml.compile` was clarified in
  its docstring.
  [(#6930)](https://github.com/PennyLaneAI/pennylane/pull/6930)
```

isaacdevlugt committed:
Apply suggestions from code review

isaacdevlugt committed:
Merge branch 'master' into compile-docstring

AntonNI8 reviewed (approved):
Nice!

AntonNI8 commented on doc/releases/changelog-dev.md:
```diff
@@ -244,6 +244,10 @@
 
 <h3>Documentation 📝</h3>
 
+* The definitition of the `pipeline` argument for `qml.compile` was clarified in
```
```suggestion
* The definition of the `pipeline` argument for `qml.compile` was clarified in
```

AntonNI8 committed:
fix typo

isaacdevlugt committed:
Merge branch 'master' into compile-docstring

runora95 committed:
update branch references for master to main (#9128)

**Context:**
Pennylane will be changing the default branch from `master` to `main`.
All the references of master in the workflows and documentation will
need to be changed to main. This PR will be merged into main after the
default branch is changed.

**Description of the Change:**
Changing references of `master` to `main`. 

**Benefits:**
Workflow and documentation references will be changed to the new default
branch.

**Possible Drawbacks:**
None

**Related GitHub Issues:**
https://github.com/PennyLaneAI/pennylane/pull/9127

ringo-but-quantum committed:
[no ci] bump nightly version

dwierichs committed:
Implement quantum part of sum-of-Slaters state preparation (#8997)

**Context:**
We want to implement [sum of slaters state
prep](https://journals.aps.org/prxquantum/abstract/10.1103/PRXQuantum.5.040339).
This method consists of a quantum circuit using dense state prep, QROM
and bit encodings.
In #8964 we add the classical preprocessing required for the bit
encodings.

**Description of the Change:**
Adds the quantum circuit implementation of the state preparation

**Benefits:**
New sparse state prep

**Possible Drawbacks:**

**Related GitHub Issues:**

---------

Co-authored-by: Korbinian Kottmann <43949391+Qottmann@users.noreply.github.com>
Co-authored-by: Mudit Pandey <18223836+mudit2812@users.noreply.github.com>
Co-authored-by: Jay Soni <jbsoni@uwaterloo.ca>
Co-authored-by: Isaac De Vlugt <34751083+isaacdevlugt@users.noreply.github.com>

albi3ro committed:
Basis rotation as a subroutine (#9026)

**Context:**

We are starting to port over our templates to be "Subroutines" where
they are defined around a qfunc definition that is compatible both
tape-based PennyLane and Catalyst. This is the first one to be ported
over, and others should follow soon.

**Description of the Change:**

Switches from `BasisRotation` being implemented as an `Operator` to
being a `Subroutine`.

**Benefits:**

`BasisRotation` can now be used with program capture Catalyst, unlocking
important ftqc and researcher workflows, as it preserves structured
control flow all the way down.

**Possible Drawbacks:**

We are still finding consequences of this change.

**Related GitHub Issues:**

[sc-110650]

---------

Co-authored-by: Astral Cai <astral.cai@xanadu.ai>
Co-authored-by: Marcus Edwards <94946848+comp-phys-marc@users.noreply.github.com>
Co-authored-by: Mudit Pandey <18223836+mudit2812@users.noreply.github.com>
Co-authored-by: David Wierichs <david.wierichs@xanadu.ai>
Co-authored-by: Marcus Edwards <marcusstephenedwards@gmail.com>

andrijapau committed:
feat: link `dynamic_one_shot` and `split_to_single_terms` transforms to their passes (#9129)

**Context:**

We added two new passes to Catalyst and we want to "link" them to their
PL counterparts using the new unified API.

**Description of the Change:**

Simply add `pass_name=` for both and record any differences with `QJIT`.

[sc-103777]

---------

Co-authored-by: Isaac De Vlugt <34751083+isaacdevlugt@users.noreply.github.com>

cvqmm committed:
Optax QSP Iterative Solver (#8685)

**Context:** Use Optax optimizers instead of Scipy's

**Description of the Change:**

**Benefits:** JAX friendly & whole optimization loop can be jit-compiled

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-110370] [sc-110735]

---------

Co-authored-by: ringo-but-quantum <>
Co-authored-by: Yushao Chen (Jerry) <chenys13@outlook.com>
Co-authored-by: Guillermo Alonso-Linaje <65235481+KetpuntoG@users.noreply.github.com>
Co-authored-by: Diego <67476785+DSGuala@users.noreply.github.com>
Co-authored-by: Ali Asadi <10773383+maliasadi@users.noreply.github.com>
Co-authored-by: Astral Cai <astral.cai@xanadu.ai>
Co-authored-by: ringo-but-quantum <github-ringo-but-quantum@xanadu.ai>
Co-authored-by: Christina Lee <chrissie.c.l@gmail.com>
Co-authored-by: Diego <diego_guala@hotmail.com>

mudit2812 committed:
Remove old `pytest-benchmark` infra (#7900)

`pytest-benchmark` was used before the PL benchmarks suite was set up.
There are remnants of the old benchmark workflows in the source code,
which this PR removes.

[sc-95638]

---------

Co-authored-by: Andrija Paurevic <46359773+andrijapau@users.noreply.github.com>

ringo-but-quantum committed:
[no ci] bump nightly version

jzaia18 committed:
Prevent pass-by-pass specs from outputting initial tape data when only MLIR passes are present (#9091)

**Context:**
The level numbering for pass-by-pass currently includes "Before (tape)
transforms" and "Before MLIR passes" as separate levels. This can be
confusing or misleading when the user does not actually apply any tape
transforms since the 2 "before" steps are located at levels 0 and 1,
leaving the first real user level at position 2.

Moreover, the "Before transforms" level forces the program to be
unrolled into a tape. When using program capture this can produce
misleading results, since the conversion into a tape may alter how
resources are collected, leading to unexpected values.

**Description of the Change:**
* Renames "Before Transforms" to "Before Tape Transforms"
* Removes the "Before Tape Transforms" level from the output of
pass-by-pass specs _when there are no tape transforms_. When at least 1
tape transform exists, the output will continue to show an inserted MLIR
lowering pass at the appropriate level.
* Various minor refactors to improve specs maintainability

**Benefits:**
* More straightforward results, less confusion about how level number
maps onto a user compilation pipeline
* Improved performance when there are no tape transforms (unrolling into
a tape can be expensive)

**Possible Drawbacks:**
The output of pass-by-pass `specs` between version 0.44 and 0.45 will in
some cases be quite different

**Related GitHub Issues:**
[sc-112200]
Builds atop changes made in #9088 
Integration tests for this update are in
https://github.com/PennyLaneAI/catalyst/pull/2533

---------

Co-authored-by: Andrija Paurevic <46359773+andrijapau@users.noreply.github.com>
Co-authored-by: ANT0N <39093564+AntonNI8@users.noreply.github.com>

Qottmann committed:
Binary representation utility function (#9117)

### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
If you've fixed a bug or added code that should be tested, add a test to
the
      test directory!

- [ ] All new functions and code must be clearly commented and
documented.
If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file,
summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.


------------------------------------------------------------------------------------------------------------

**Context:**

This is used a lot when discretizing rotation gates to quantum
arithmetic

**Description of the Change:**

new function

**Benefits:**

avoids duplication, useful docstring, correct rounding

**Possible Drawbacks:**

none

**Related GitHub Issues:**

---------

Co-authored-by: David Wierichs <david.wierichs@xanadu.ai>
Co-authored-by: Guillermo Alonso-Linaje <65235481+KetpuntoG@users.noreply.github.com>

github-actions[bot] committed:
Update test durations (#9147)

Automatic update of test duration files

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Yushao Chen (Jerry) <chenys13@outlook.com>

ringo-but-quantum committed:
[no ci] bump nightly version

albi3ro committed:
Capture value and grad (#8814)

**Context:**

We are trying to unify the pennylane and catalyst frontends into a
single plxpr pipeline.

**Description of the Change:**

Adds a `qml.value_and_grad` function that works with both catalyst
frontends.

**Benefits:**

Users can efficiently calculate the results and gradient at the same
time.

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-106504]

---------

Co-authored-by: Yushao Chen (Jerry) <chenys13@outlook.com>
Co-authored-by: David Wierichs <david.wierichs@xanadu.ai>

JerryChen97 committed:
Add changelog for #8685 (#9145)

ringo-but-quantum committed:
[no ci] bump nightly version

isaacdevlugt committed:
mergE
